makemake.sh

```diff
@@ -78,25 +89,37 @@ if ! command -v $MAKE >/dev/null; then
 	echo "On Ubuntu and Debian run: 'sudo apt-get update' and 'sudo apt-get install -y build-essential'" >&2
 	exit 1
 fi
-if [[ -n $CC ]]; then
-	if ! command -v "$CC" >/dev/null; then
-		echo "Error: $CC is not installed." >&2
-		exit 1
-	fi
-elif ! command -v gcc >/dev/null; then
-	echo "Error: This script requires the GNU C compiler" >&2
-	echo "On Ubuntu and Debian run: 'sudo apt-get update' and 'sudo apt-get install -y build-essential'" >&2
+if ! command -v "$CC" >/dev/null; then
+	echo "Error: $CC is not installed." >&2
+	echo "For GCC, on Ubuntu and Debian run: 'sudo apt-get update' and 'sudo apt-get install -y build-essential'" >&2
 	exit 1
 fi
 
+MAKE_ARGS+=(-j "$CPU_THREADS")
+
+CCVER=$($CC --version | head -n1)
+if [[ $CCVER == *"clang"* ]]; then
+	echo "Clang detected as C compiler. Enabling ThinLTO for multithreaded codegen."
+	C_ARGS_LTO+=(-flto=thin)
+elif [[ $CCVER == *"gcc"* ]]; then
+	echo "GCC detected as C compiler. Enabling multithreaded LTO."
+	C_ARGS_LTO+=(-flto="$CPU_THREADS")
```

Comment:
I do not believe we would want ThinLTO for Clang, as it provides less performance. Also, `-flto=auto` should work with GCC:
```suggestion
if [[ $CCVER == *clang* ]]; then
	echo "Clang detected as C compiler. Enabling multithreaded LTO."
	C_ARGS_LTO+=(-flto="$CPU_THREADS")
elif [[ $CCVER == *gcc* ]]; then
	echo "GCC detected as C compiler. Enabling multithreaded LTO."
	C_ARGS_LTO+=(-flto=auto)
```

Reply:
As for why not =n on clang: clang 8.4.0 has =thin but not =n, nor any partitioning options. Yes I'm for nudging the user towards getting a newer clang, but I'm also a bit of a coward…